core/mix/ensure_gear_dependencies.ex

```diff
@@ -13,22 +13,41 @@ defmodule Mix.Tasks.Compile.EnsureGearDependencies do
     Mix.Project.config()
     |> Keyword.fetch!(:antikythera_gear)
     |> Keyword.fetch!(:gear_deps)
-    |> Enum.each(fn dep -> ensure_gear_dependency!(elem(dep, 0)) end)
+    |> Enum.each(fn dep -> ensure_gear_dependency!(elem(dep, 0), extract_opts_from_dep(dep)) end)
   end
 
-  defp ensure_gear_dependency!(gear_name) do
-    gear_mixfile_module = Mix.Project.in_project(gear_name, gear_dir(gear_name), fn mod -> mod end)
+  defp extract_opts_from_dep({_app, _req, opts})             , do: opts
+  defp extract_opts_from_dep({_app, opts}) when is_list(opts), do: opts
+  defp extract_opts_from_dep({_app, _req})                   , do: []
+
+  defp ensure_gear_dependency!(gear_name, opts) do
+    gear_mixfile_module = Mix.Project.in_project(gear_name, gear_dir(gear_name, opts), fn mod -> mod end)
     if gear_mixfile_module.project() |> Keyword.has_key?(:antikythera_gear) do
       :ok
     else
       Mix.raise("#{gear_name} is not a gear application! `gear_deps/0` may only contain gear dependencies.")
     end
   end
 
-  defp gear_dir(gear_name) do
-    cwd = File.cwd!()
+  defp gear_dir(gear_name, opts) do
+    case Keyword.fetch(opts, :path) do
+      {:ok, path} -> gear_dir_as_local_dep(path)
+      :error      -> gear_dir_as_remote_dep(gear_name)
```

Comment:
`gear_dir_as_path_dep`, `gear_dir_as_non_path_dep`
could sound nicer for these functions in the sense that the term "path dependency" is used by official docs such as [`Mix.Tasks.Deps`](https://hexdocs.pm/mix/Mix.Tasks.Deps.html).